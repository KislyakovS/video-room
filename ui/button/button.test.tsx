import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Button from './button';

describe('Component Button:', () => {
    it('snapshot', () => {
        const { container } = render(<Button>Button</Button>);

        expect(container).toMatchSnapshot();
    });

    it('should call for a click on the button', async () => {
        const user = userEvent.setup();
        const onClick = jest.fn();

        const { getByRole } = render(<Button onClick={onClick}>Button</Button>);
        const button = getByRole('button');

        await user.click(button);

        expect(onClick).toHaveBeenCalled();
    });
})