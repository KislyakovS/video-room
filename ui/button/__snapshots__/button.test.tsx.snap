// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button: snapshot 1`] = `
<div>
  <button
    class="button"
  >
    Button
  </button>
</div>
`;
